lib/src/main/scala/spinal/lib/Stream.scala

```diff
@@ -906,13 +956,19 @@ class StreamCCByToggle[T <: Data](dataType: T, inputClock: ClockDomain, outputCl
   }
 }
 
-object StreamDispatcherSequencial{
+/**
+ * This is equivalent to a StreamDemux, but with a counter attached to the port selector.
+ */
+// TODO is there any reason to have a proper class for this if we simply could reuse StreamDemux instead? ~piegames
```

Comment:
That's likely right, could hav eused StreamDemux with some proper logic around it :)

Reply:
Can I change this, then?

Reply:
Sure ^^